### PR TITLE
Prevent events with return (2x)

### PR DESCRIPTION
**Pull request by shargon**
Port of https://github.com/neo-project/neo-devpack-dotnet/pull/124 to 2x

Wait for: https://github.com/neo-project/neo-devpack-dotnet/pull/122

### CODE REVIEW

File: Neo.Compiler.MSIL/NeoModule.cs [hunk 1]
```diff
@@ -147,7 +149,19 @@ public class NeoEvent
         public string name;
         public string displayName;
         public List<NeoParam> paramtypes = new List<NeoParam>();
-        public string returntype;
```
Comment (lock9) [inline Neo.Compiler.MSIL/NeoModule.cs]: I think we should keep the return type for Neo 2, for compatibility purposes.

**Comment by lapetina9** [comment]
## Neo PR Test Report

**Repository:** neo-devpack-dotnet  
**Number:** 125  
**Title:** Prevent events with return (2x)  
**Change description:** Port of #124 to 2x


## Code version / branch

**Neo-devpack-dotnet:**  shargon:close-123-2x


## Test Results

| Test Name | Network | OS | Test Result | Comments |
| ------- |:-------:| --:| -----------:| --------:|
| Initial UT | UT | Localhost | OK | 
| Additional unit test to delegate event with arguments | UT | Localhost | OK | |

## Additional info
The code works but removes a feature unecessarily. We recommend putting back the return type, even if it is always Void.
We merged #122 to test these changes.

## **Test List**

**1. Initial UT**  
Run solution unit tests.

**2. Additional unit test to event with return type**  
  
Create Smart Contract with event return type int.

Expected result:
- While reading Smart Contract, engine throws a NotSupportedException.

**Comment by shargon** [comment]
> Put the return type back in the ABI, even when it is void. I can't see gains for this change in Neo 2.

This pr is precisely for this hahha

**Comment by lock9** [comment]
Hi @shargon , but why are we removing the property? Isn't the purpose of this PR block events with return type *different* than void? 
I think that removing a property is a breaking change, I thought that the idea of this PR was to prevent invalid events, not to remove them.
What is your opinion? Do you think we really should remove it or just allow void return types?

**Comment by shargon** [comment]
> Isn't the purpose of this PR block events with return type different than void?

The purpose is throw an error when one event return different than void, because is not in NEP3

**Comment by lock9** [comment]
Hi @shargon,  can't we do that without changing the JSON file? We should keep the exception.
I'm going to remove my request for changes. Since #122 is not merged yet, there will be no difference.

**Comment by shargon** [comment]
Why? it's a new compilation, why you need to compile a wrong SC?

File: Neo.Compiler.MSIL/NeoModule.cs (hunk 1 above)
Comment (shargon) [reply Neo.Compiler.MSIL/NeoModule.cs]: Is not a breaking change, in the case that you want to recompile your SC, you need to fit to the new rules, but your SC still working on the chain.

**Comment by shargon** [comment]
Travis still broken 😔

**Comment by erikzhang** [comment]
Please re-create the PR from a new branch in neo-project. Don't create PRs from personal branches. Then we can restart travis easily.

**Comment by cloud8little** [comment]
> Travis still broken 😔

There is no .travis.yml file for branch: master-2.x.

**Comment by shargon** [comment]
> Don't create PRs from personal branches.

The problem is that i can't use gitkraken with the neo-project branchs, i have no rights with gitkraken :(

**Comment by shargon** [comment]
Closed in favor of https://github.com/neo-project/neo-devpack-dotnet/pull/136

**Pull request closed without merging**